DbgProvider/public/Debugger/DbgEngDebugger.cs

```diff
@@ -2375,6 +2357,20 @@ private void _CheckMemoryReadHr( ulong address, int hr )
             CheckHr( hr );
         } // end _CheckMemoryReadHr()
 
+        private bool _TryCheckMemoryReadHr( int hr )
+        {
+            if( (HR_ERROR_READ_FAULT == hr) ||
+                (HR_STATUS_NO_PAGEFILE == hr) ||
+                (E_NOINTERFACE == hr) ||
+                (HR_STATUS_NO_MORE_ENTRIES == hr) )
```

Comment:
>HR_STATUS_NO_MORE_ENTRIES [](start = 17, length = 25)

Is this what is returned if `ReadPointersVirtual` can't read that many pointers?

Reply:
The only place that I remember for sure returns that is SearchVirtual; I'm not sure if there's anything else that does return it. 

It looks like ReadPointersVirtual just passes through the ReadVirtual HRESULT when it entirely fails to read a pointer, and E_FAIL when it only partially reads a pointer. (Also it looks like DbgShell's E_FAIL has a transposed 4 & 0)

Reply:
Oy... looks like that `E_FAIL` is `OLE_E_NOTRUNNING`; oops. Now I wonder if I actually saw 80040005... doesn't seem likely, though; we should just fix that transposition. Thanks for finding that.

---
In reply to: [259683714](https://github.com/Microsoft/DbgShell/pull/75#discussion_r259683714) [](ancestors = 259683714)